app/src/test/java/org/mozilla/fenix/components/toolbar/DefaultBrowserToolbarControllerTest.kt

```diff
@@ -100,6 +104,11 @@ class DefaultBrowserToolbarControllerTest {
     fun setUp() {
         MockKAnnotations.init(this)
 
+        mockkStatic("org.mozilla.fenix.perf.PerfNavControllerKt")
+        every { waitForNavGraphInflation(any()) } returns Unit
```

Comment:
I'm not sure this is the most correct solution – it's essentially saying, "this method is problematic so we stop calling it" by using mocking, which is a fairly fragile API and is decoupled from the actual implementation of the code.

For example, a more robust solution might inject a testing-only implementation of some class where we don't look in the weak map since we know we're in testing mode and it's not expected that we'd run that.

I don't know if there is a better solution but I wanted to highlight it as something to think about. In general, if "how can I test this code" and "how do I prevent unrelated tests from breaking" is an after thought, it's a smell that there might be more robust ways to design the code, which usually pushes you towards better encapsulation.